### PR TITLE
fix: [CDS-73771]: Adds support for multi service propagation

### PR DESCRIPTION
**Pull request by samarth-sharma31**
### Summary

Service propagation -> The service information of any stage can be propagated from a previous stage given these two share similar deployment configuration (like deployment type).
Earlier this support was only present for single service stages - Now this PR extends it to multiple services.

#### Screenshots
Screenshot 2023-07-31 at 12 20 57 AM

Screenshot 2023-07-31 at 12 16 15 AM

Stage deletion test case and nested propagation scenario -->

https://github.com/harness/harness-core-ui/assets/96409598/2889dce3-9820-4754-a363-13071d632781

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/services/cd-ng/index.tsx [hunk 1]
```diff
@@ -18417,9 +18417,11 @@ export interface ServicesV2YamlMetadataDTO {
   serviceV2YamlMetadataList?: ServiceV2YamlMetadata[]
 }
 
+// TODO: To be removed post development
 export interface ServicesYaml {
   metadata?: ServicesMetadata
   values?: ServiceYamlV2[]
+  useFromStage?: ServiceUseFromStageV2
```
Comment (chetanvekariya-harness) [inline src/services/cd-ng/index.tsx:18424]: Is this manual change? We can create our own type to add this new field and extend this interface.

Comment (samarth-sharma31) [reply src/services/cd-ng/index.tsx]: Removed, not required anymore.

File: src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/DeployServiceEntityWidget.tsx [hunk 2]
```diff
@@ -146,7 +147,7 @@ export default function DeployServiceEntityWidget({
   }, [])
 
   useDeepCompareEffect(() => {
-    if (setupModeType === setupMode.PROPAGATE) {
+    if (setupModeType === setupMode.PROPAGATE || setupModeType === setupMode.DIFFERENT) {
```
[the remark this answers is not in the document]
Comment (samarth-sharma31) [reply src/modules/75-cd/components/PipelineSteps/DeployServiceEntityStep/DeployServiceEntityWidget.tsx:150]: yes agreed, simplified now

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by vikrant-harness — APPROVED** [review]

**Comment by samarth-sharma31** [comment]
Ticket for tracking coverage -> https://harness.atlassian.net/browse/CDS-79015

**Review by bhavya-sinha — APPROVED** [review]

**Pull request merged**